skt/executable.py

```diff
@@ -312,72 +312,104 @@ def cmd_merge(args):
 
 
 @junit
-def cmd_build(cfg):
+def cmd_build(args):
     """
     Build the kernel with specified configuration and put it into a tarball.
 
     Args:
-        cfg:    A dictionary of skt configuration.
+        args:    Command line arguments
     """
     global retcode
     tstamp = datetime.datetime.strftime(datetime.datetime.now(),
                                         "%Y%m%d%H%M%S")
 
     tgz = None
     builder = KernelBuilder(
-        source_dir=cfg.get('workdir'),
-        basecfg=cfg.get('baseconfig'),
-        cfgtype=cfg.get('cfgtype'),
-        extra_make_args=cfg.get('makeopts'),
-        enable_debuginfo=cfg.get('enable_debuginfo'),
-        rh_configs_glob=cfg.get('rh_configs_glob'),
-        localversion=cfg.get('localversion')
+        source_dir=full_path(args.get('workdir')),
```

Comment:
Actually we are using relative paths on the reporter :smirk: . Maybe you want to take advantage of you're touching this part of the code to simplify the reporter code later :).

Reply:
Valid point. I'll remove it.